physlr/physlr.py

```diff
@@ -1674,62 +1799,141 @@ def physlr_molecules(self):
+                Physlr.set_settings(round)
+            # Physlr.args.prune_bridges = 10
+            # Physlr.args.prune_branches = 10
+            # Physlr.args.prune_junctions = 50
+            if round > 1:
+                print(
+                    int(timeit.default_timer() - t0),
+                    "Detecting junction-causing barcodes",
+                    file=sys.stderr)
+                junctions = Physlr.identify_junctions_graph(gin)
+            # round = round + 1
+            if junctions:
+                print(
+                    int(timeit.default_timer() - t0),
+                    "Working on the junction-causing barcodes with algs:", alg_list,
+                    file=sys.stderr)
+                if "sqcos" in alg_list or "cos" in alg_list or \
+                        "sqcosbin" in alg_list or "cosbin" in alg_list:
+                    print(
+                        int(timeit.default_timer() - t0),
+                        "cost:", Physlr.args.cost, "sqcost:", Physlr.args.sqcost,
+                        file=sys.stderr)
+                nodes_to_process = junctions
+            else:
+                print(
+                    int(timeit.default_timer() - t0),
+                    "Working on the whole overlap graph with algs:", alg_list,
+                    file=sys.stderr)
+                nodes_to_process = gin
+            if self.args.threads == 1:
+                molecules = dict(
+                    self.determine_molecules(
+                        gin, u, junctions, alg_list) for u in progress(nodes_to_process))
+            else:
+                Physlr.graph = gin
+                Physlr.junctions = junctions
+                Physlr.args.strategy = alg_list
+                with multiprocessing.Pool(self.args.threads) as pool:
+                    molecules = dict(pool.map(
+                        self.determine_molecules_process,
+                        progress(nodes_to_process), chunksize=100))
+                Physlr.graph = None
+            print(int(timeit.default_timer() - t0), "Identified molecules", file=sys.stderr)
+            if not molecules:
+                print(int(timeit.default_timer() - t0), "Identidied no molecules", file=sys.stderr)
+
+            # Add vertices.
+            if round > 1:
+                gout = gin.copy()
+            else:
+                gout = nx.Graph()
+            cumul_nmolecules = 0
+            cumul_junctions = 0
+            for u, vs in sorted(molecules.items()):
+                n = gin.nodes[u]["n"]
+                nmolecules = 1 + max(vs.values()) if vs else 0
+                if nmolecules > 1:
+                    cumul_nmolecules += (nmolecules - 1)
+                    cumul_junctions += 1
+                for i in range(nmolecules):
+                    if round > 1:
+                        gout.add_node(f"{u}-{i}", n=n)
+                    elif round == 1:
+                        gout.add_node(f"{u}_{i}", n=n)
```

Comment:
What is this if else loop for?

Reply:
for rounds 2+ we add a symbol other than `_`, here its `-`. I wanted to discuss this with you. Is dash confusing for some datasets?

Reply:
My apologies, they looked the same when I was reviewing.